src/MemcachedCachePool.php

```diff
@@ -45,7 +49,11 @@ protected function clearAllObjectsFromCache()
 
     protected function clearOneObjectFromCache($key)
     {
-        if ($this->cache->delete($key)) {
+        $keyString = $this->getHierarchyKey($key, $path);
```

Comment:
I think you have forgotten to do `$this->commit()` to make sure to move all your items in the deferred queue to the Memcached storage.